dm/syncer/error.go

```diff
@@ -211,13 +212,66 @@ func (s *Syncer) handleSpecialDDLError(tctx *tcontext.Context, err error, ddls [
 	}
 	// TODO: add support for downstream alter pk without schema
 
+	// it handles the operations of DDL when encountering `invalid connection`
+	invalidConnF2 := func(tctx *tcontext.Context, err error, ddls []string, index int, conn *dbconn.DBConn, createTime int64) error {
+		if len(ddls) == 0 || index > len(ddls)-1 || errors.Cause(err) != mysql.ErrInvalidConn || createTime == -1 {
+			return err // return the original error
+		}
+
+		statusChan := make(chan string)
+		var status string
+		var err2 error
+
+		for {
+			status, err2 = getDDLStatusFromTiDB(tctx, conn, ddls[index], createTime) // status is synced
+			failpoint.Inject("TestStatusRunning", func() {
+				status = "running"
+			})
+			failpoint.Inject("TestStatusCancelled", func() {
+				status = "cancelled"
+			})
+			failpoint.Inject("TestStatusQueueing", func() {
+				status = "queueing"
+			})
+			failpoint.Inject("TestStatusNone", func() {
+				status = "none"
+			})
+			if err != nil {
+				return err2
+			}
+			statusChan <- status
+			select {
+			case <-tctx.Ctx.Done():
+				return nil
+			case status = <-statusChan:
+				switch status {
+				case "running":
+				case "rollingback":
+				case "rollback done":
+					return nil
+				case "done":
+					return nil
+				case "cancelled":
+					return err
```

Comment:
should not return `err` because `err` is "invalid connection", we should return something like "DDL %s executed in background and met error". And can we find the error from TiDB?